Microsoft.DotNet.Interactive.PowerShell/Host/DollarProfileHelper.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Management.Automation;
+using System.Reflection;
+
+namespace Microsoft.DotNet.Interactive.PowerShell.Host
+{
+    internal static class DollarProfileHelper
+    {
+        public delegate string GetFullProfileFileNameDelegate(
+            string hostId,
+            bool forCurrentUser);
+
+        private static Lazy<GetFullProfileFileNameDelegate> s_LazyGetFullProfileFileName =
+            new Lazy<GetFullProfileFileNameDelegate>(() =>
+            {
+                // Grab GetFullProfileName static method which handles a lot of the profile
+                // path generation.
+                MethodInfo method = typeof(HostUtilities).GetMethod(
```

Comment:
It should be relatively straightforward to do the same without reflection.
The current user profile base directory can be got by `Platform.SelectProductNameForDirectory(Platform.XDG_Type.CONFIG)`.
The all user profile base directory is `$PSHome` in PowerShell, namely `Path.GetDirectoryName(typeof(PSObject).Assembly.Location)`.

Maybe the `GetFullProfileFileName` should be made public, but before that I think it's better we just depend on the public APIs (good code practice for other hosting application to copy).

Reply:
@daxian-dbw `Platform.SelectProductNameForDirectory(Platform.XDG_Type.CONFIG)` is non-Windows only.

Reply:
If you happen to know where I can get the CurrentUserCurrentHost path on Windows, I can make this change. I have the rest:

```csharp
 public static string GetFullProfileFilePath(bool forCurrentUser)
        {
            if (!forCurrentUser)
            {
                string pshome = Path.GetDirectoryName(typeof(PSObject).Assembly.Location);
                return Path.Combine(pshome, _profileName);
            }

            if (Platform.IsWindows)
            {
                // GET CURRENTUSERCURRENTHOST ON WINDOWS
            }
            else
            {
                string configPath = Platform.SelectProductNameForDirectory(Platform.XDG_Type.CONFIG);
                return Path.Combine(configPath, _profileName);
            }
        }
```

Reply:
Looks like it's here:
https://github.com/PowerShell/PowerShell/blob/b1e998046e12ebe5da9dee479f20d479aa2256d7/src/System.Management.Automation/CoreCLR/CorePsPlatform.cs#L143-L150

Reply:
Bummer. Some internal methods should be made public to improve the dev experience, such as `Platform.ConfigDirectory` and `Platform.CacheDirectory`.

Reply:
I opened an issue about this... but I doubt it'll be addressed any time soon:
https://github.com/PowerShell/PowerShell/issues/12322